### ISSUE TITLE
`prefer-type-literal-last`: exclude from `unopinionated`, relax some checks

### ISSUE THREAD

**Issue by andreww2012** [sindresorhus/eslint-plugin-unicorn#3313]
### Description

Typescript union/intersection members' order may depend on project conventions, the code author's preferences, be put in randomly, etc. Additionally, as the rule description correctly notes, order in intersections matters, yet it's still reported by default. I believe these two reasons are strong enough to warrant an exclusion of this rule from the unopinionated preset.

Additionally, I want to point out two more things about it:
1. May an option be added to disable (or explicitly enable, if otherwise disabled by default) intersection checking?
1. Should these cases actually be reported?
   ```ts
   type Test = {foo: string} & ({bar: number} | {baz: boolean});
             // ^^^^^^^^^^^^^
   type Test2 = {foo: string} | ({bar: number} & {baz: boolean});
             // ^^^^^^^^^^^^^
   ```
   Changing the order here IMHO doesn't have any effect on the discoverability of any of these types.

### Examples

N/A
